marketdata/market/coingecko/coingecko.go

```diff
@@ -0,0 +1,94 @@
+package coingecko
+
+import (
+	"github.com/trustwallet/blockatlas/marketdata/coingecko"
+	"github.com/trustwallet/blockatlas/marketdata/market"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"strings"
+)
+
+const (
+	id = "coingecko"
+)
+
+type Market struct {
+	client *coingecko.Client
+	cache  *coingecko.Cache
+	market.Market
+}
+
+func InitMarket(api string, updateTime string) market.Provider {
+	m := &Market{
+		client: coingecko.NewClient(api),
+		Market: market.Market{
+			Id:         id,
+			UpdateTime: updateTime,
+		},
+	}
+	return m
+}
+
+func (m *Market) GetData() (result blockatlas.Tickers, err error) {
```

Comment:
```suggestion
func (m *Market) GetData() (blockatlas.Tickers, error) {
```

Reply:
This doesn't work here as I use named return statement

Reply:
Don't mix the implicit return variables and the standard return is confusing

You are using only to handle the error, don't make sense create a variable global for all method to use only if have an error. Because you are not using that in the [line 40](https://github.com/trustwallet/blockatlas/pull/648/files#diff-66d5ab40d52392088915d9460e055f2cR40)

this implicit return variable works well if you need to unmarshal the value inside the variables or for append slice or map

Reply:
fixed [line 40](https://github.com/trustwallet/blockatlas/pull/648/files#diff-66d5ab40d52392088915d9460e055f2cR40)